idaes/gas_distribution/unit_models/pipeline.py

```diff
@@ -0,0 +1,394 @@
+        default=None,
+        domain=int,
+        description="Number of collocation points",
+        doc="""Number of collocation points to use (equivalent to Pyomo ncp
+argument)."""))
+
+    def build(self):
+        super(GasPipelineData, self).build()
+
+        # self.config is the ConfigBlock "instantiated" from self.CONFIG
+        # in ProcessBlockData.
+        config = self.config
+        time = self.flowsheet().time
+        property_package = config.property_package
+
+        if len(property_package.phase_list) != 1:
+            raise ValueError(
+                "%s can only be constructed with a "
+                "single-phase property package.\n"
+                "Got phases %s."
+                % (self.__class__, [p for p in property_package.phase_list])
+            )
+        self.phase = next(iter(property_package.phase_list))
+        if self.phase != "Vap":
+            raise ValueError(
+                "%s can only be constructed with a single phase, \"Vap\"."
+                "Got phase %s."
+                % (self.__class__, self.phase)
+            )
+
+        #
+        # Make sure property package has the properties that this unit
+        # model requires.
+        #
+        property_dict = property_package.get_metadata().properties
+        if "pressure" not in property_dict:
+            raise ValueError(
+                "Property package supplied to pipeline must have a property "
+                "for 'pressure', which was not found in %s."
+                % type(property_package)
+            )
+
+        #
+        # Set up control volume:
+        #
+        #cv_config = {
+        #    "transformation_method": config.transformation_method,
+        #    "transformation_scheme": config.transformation_scheme,
+        #    "finite_elements": config.finite_elements,
+        #    "collocation_points": config.collocation_points,
+        #    "has_holdup": config.has_holdup,
+        #}
+        cv_config = config()
+        self.control_volume = ControlVolume1DBlock(default=cv_config)
+        self.control_volume.add_geometry()
+        self.control_volume.add_state_blocks(
+            has_phase_equilibrium=False,
+        )
+        self.control_volume.add_phase_component_balances()
+
+        cv = self.control_volume
+        self.add_diameter()
+        self.add_friction_factor()
+
+        #
+        # Add constraints indexed by space before applying transformation
+        #
+        # HACK: speed of sound should be a property of the state block.
+        # But the generic property ideal EOS does not have a property
+        # for speed of sound. This should probably be fixed, but we
+        # will have to develop our own property package anyway to
+        # account for the non-ideal behavior of the natural gas.
+        # For now, we just add speed of sound to the unit model so we can
+        # calculate some values.
+        #self.add_speed_of_sound()
```

Comment:
Is this outdated? I saw the speed of sound on the property package instead of here.